Run unit-tests on GHA

kraenhansen opened the pull request:
## Summary:

As suggested by [this comment](https://github.com/facebook/react-native/pull/45133#issuecomment-2571621818), https://github.com/facebook/react-native/pull/45133 inadvertently disabled unit tests on GHA and I believe these changes will re-enable running them.

To actually make those green again, I believe https://github.com/facebook/react-native/pull/48498 needs to merge first.

## Changelog:

[INTERNAL] [FIXED] - Start running unit tests on GHA again.

## Test Plan:

I'll await the GHA logs 🤞

kraenhansen committed:
Run unit-tests on GHA

cipolleschi commented on .github/workflows/test-all.yml:
```diff
@@ -185,7 +185,7 @@ jobs:
           jsengine: ${{ matrix.jsengine }}
           architecture: ${{ matrix.architecture }}
-          run-unit-tests: "false"
+          run-unit-tests: true
```
the problem with this is that we will run unit tests on ALL the matrix configuration. This is a bit of an overkill and it will increase the costs of CI sensibly.

I believe it would be better to copy this job in the workflow and to run unit tests with the Hermes, New Architecture, Static Library configuration only. This should be enough.

kraenhansen replied:
> copy this job in the workflow

Instead of making a copy, I'd suggest adding this as a parameter to the matrix and [use the `include` feature](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/running-variations-of-jobs-in-a-workflow#expanding-or-adding-matrix-configurations) to run it only on the configurations you mentioned.

Would you be okay with keeping things simple for now and ☝️ to be a separate PR?

cipolleschi replied:
can you move this to matrix in this pr?
This is not pretty complex, and adding a param to the matrix should not add a lot of noise.

kraenhansen replied:
> Hermes, New Architecture, Static Library configuration only

For both `Debug` and `Release` or just one of these?

cipolleschi reviewed (changes requested):
Thanks for working on this. I see that the CI jobs are red, so this probably need some more work. If they are red because they depend on #48498, you can change the destination branch to be the other PR:

<img width="826" alt="Screenshot 2025-01-06 at 10 07 24" src="https://github.com/user-attachments/assets/6184441a-a048-491e-8205-41b00ce923f8" />

I also left a comment to reduce a bit the CI costs

kraenhansen commented:
> you can change the destination branch to be the other PR

I probably could, but then you'd be reviewing a PR in my fork.
It might be simpler for me to simply include these changes in https://github.com/facebook/react-native/pull/48498?

kraenhansen commented:
This is output from [the log of the "test_ios_rntester (Hermes, NewArch, Release)" job](https://github.com/facebook/react-native/actions/runs/12632312745/job/35196003083?pr=48499):

```
2025-01-06T14:20:54.7300260Z Attempt #1 of running tests...
2025-01-06T14:21:29.1000560Z --- xcodebuild: WARNING: Using the first of multiple matching destinations:
2025-01-06T14:21:29.1376910Z { platform:iOS, id:dvtdevice-DVTiPhonePlaceholder-iphoneos:placeholder, name:Any iOS Device }
2025-01-06T14:21:29.1383110Z { platform:iOS Simulator, id:dvtdevice-DVTiOSDeviceSimulatorPlaceholder-iphonesimulator:placeholder, name:Any iOS Simulator Device }
2025-01-06T14:21:29.1392450Z { platform:iOS Simulator, id:447B3D12-251F-459A-8F21-AD29528D1EAA, OS:17.0.1, name:iPad (10th generation) }
2025-01-06T14:21:29.1554280Z { platform:iOS Simulator, id:A50D4B50-5610-4D50-A8CF-41A2BB01584C, OS:17.2, name:iPad (10th generation) }
2025-01-06T14:21:29.1582460Z { platform:iOS Simulator, id:D6FBEFBB-59BA-4348-A561-B75E69D24935, OS:17.0.1, name:iPad Air (5th generation) }
2025-01-06T14:21:29.1590120Z { platform:iOS Simulator, id:15C80D0C-2701-450E-8989-B1D861BDB399, OS:17.2, name:iPad Air (5th generation) }
2025-01-06T14:21:29.1593880Z { platform:iOS Simulator, id:07505E70-434F-4894-B708-84F4497643B2, OS:17.0.1, name:iPad Pro (11-inch) (4th generation) }
2025-01-06T14:21:29.1599600Z { platform:iOS Simulator, id:107CB696-AB9E-4BFC-AD00-B4E6F1470A0E, OS:17.2, name:iPad Pro (11-inch) (4th generation) }
2025-01-06T14:21:29.1602690Z { platform:iOS Simulator, id:6BF7D7A9-9E8A-43D1-80EF-BD6301CB6762, OS:17.0.1, name:iPad Pro (12.9-inch) (6th generation) }
2025-01-06T14:21:29.1606420Z { platform:iOS Simulator, id:91F85002-0495-4077-95EB-138A4B31BD9F, OS:17.2, name:iPad Pro (12.9-inch) (6th generation) }
2025-01-06T14:21:29.1610080Z { platform:iOS Simulator, id:9F5160B4-4E52-4E3A-BB97-8C67539E7EC5, OS:17.0.1, name:iPad mini (6th generation) }
2025-01-06T14:21:29.1612780Z { platform:iOS Simulator, id:2841CD68-372A-4C67-BE9F-730FA7D2E190, OS:17.2, name:iPad mini (6th generation) }
2025-01-06T14:21:29.1615330Z { platform:iOS Simulator, id:F86512C4-7BD2-4E0B-954A-6B5CD96E85A3, OS:17.0.1, name:iPhone 14 }
2025-01-06T14:21:29.1619220Z { platform:iOS Simulator, id:C68FA801-0220-4C45-B7A0-61F007527C24, OS:17.2, name:iPhone 14 }
2025-01-06T14:21:29.1623150Z { platform:iOS Simulator, id:9D7E5346-AE28-4536-BDF2-6D8AC9D96666, OS:17.0.1, name:iPhone 14 Plus }
2025-01-06T14:21:29.1626770Z { platform:iOS Simulator, id:CDFA0365-5A29-469E-8A51-D04402ED69BD, OS:17.2, name:iPhone 14 Plus }
2025-01-06T14:21:29.1631260Z { platform:iOS Simulator, id:0E716B60-116E-4C3B-8DB0-313A0726A827, OS:17.0.1, name:iPhone 14 Pro }
2025-01-06T14:21:29.1633880Z { platform:iOS Simulator, id:6C0517D9-8023-4D1C-9E4E-E643F233581E, OS:17.2, name:iPhone 14 Pro }
2025-01-06T14:21:29.1636860Z { platform:iOS Simulator, id:2D291A0B-763B-4C36-995F-9E351D48D6AC, OS:17.0.1, name:iPhone 14 Pro Max }
2025-01-06T14:21:29.1640320Z { platform:iOS Simulator, id:B13FFED6-0D1F-476D-99DC-F4DE09A93B77, OS:17.2, name:iPhone 14 Pro Max }
2025-01-06T14:21:29.1646590Z { platform:iOS Simulator, id:A33305FD-6CBC-4AD4-B5E3-65CEF85B6969, OS:17.0.1, name:iPhone 15 }
2025-01-06T14:21:29.1650660Z { platform:iOS Simulator, id:116D1E30-F4F0-4737-BFDB-746F38A77AF8, OS:17.2, name:iPhone 15 }
2025-01-06T14:21:29.1653130Z { platform:iOS Simulator, id:F8DBD11C-F466-4CA4-921B-01AC9BD4BF21, OS:17.0.1, name:iPhone 15 Plus }
2025-01-06T14:21:29.1656180Z { platform:iOS Simulator, id:849D7D2A-B6B5-4878-B10E-F612D0405D83, OS:17.2, name:iPhone 15 Plus }
2025-01-06T14:21:29.1659470Z { platform:iOS Simulator, id:044DC2BD-D34B-4B49-AF4F-F11574B6D782, OS:17.0.1, name:iPhone 15 Pro }
2025-01-06T14:21:29.1662690Z { platform:iOS Simulator, id:4DE5EECA-62A1-4E24-93E9-3295810361B5, OS:17.2, name:iPhone 15 Pro }
2025-01-06T14:21:29.1665540Z { platform:iOS Simulator, id:22486642-C0C6-41D0-A3AF-C80C2DC16425, OS:17.0.1, name:iPhone 15 Pro Max }
2025-01-06T14:21:29.1667970Z { platform:iOS Simulator, id:2A4BA5E4-684C-4336-9CA1-5BB2F27CA9A7, OS:17.2, name:iPhone 15 Pro Max }
2025-01-06T14:21:29.1671080Z { platform:iOS Simulator, id:FB316C94-655B-4F52-94EF-75AEC7ABAAD7, OS:17.0.1, name:iPhone SE (3rd generation) }
2025-01-06T14:21:29.1674560Z { platform:iOS Simulator, id:D83FEAB7-55E8-49C8-9E5D-697943752185, OS:17.2, name:iPhone SE (3rd generation) }
2025-01-06T14:21:38.9324150Z ** BUILD FAILED **
2025-01-06T14:21:38.9343670Z 
2025-01-06T14:22:09.1228920Z --- xcodebuild: WARNING: Using the first of multiple matching destinations:
2025-01-06T14:22:09.2236420Z { platform:iOS, id:dvtdevice-DVTiPhonePlaceholder-iphoneos:placeholder, name:Any iOS Device }
2025-01-06T14:22:09.2265470Z { platform:iOS Simulator, id:dvtdevice-DVTiOSDeviceSimulatorPlaceholder-iphonesimulator:placeholder, name:Any iOS Simulator Device }
2025-01-06T14:22:09.3016920Z { platform:iOS Simulator, id:447B3D12-251F-459A-8F21-AD29528D1EAA, OS:17.0.1, name:iPad (10th generation) }
2025-01-06T14:22:09.3021930Z { platform:iOS Simulator, id:A50D4B50-5610-4D50-A8CF-41A2BB01584C, OS:17.2, name:iPad (10th generation) }
2025-01-06T14:22:09.3029960Z { platform:iOS Simulator, id:D6FBEFBB-59BA-4348-A561-B75E69D24935, OS:17.0.1, name:iPad Air (5th generation) }
2025-01-06T14:22:09.3033820Z { platform:iOS Simulator, id:15C80D0C-2701-450E-8989-B1D861BDB399, OS:17.2, name:iPad Air (5th generation) }
2025-01-06T14:22:09.3037610Z { platform:iOS Simulator, id:07505E70-434F-4894-B708-84F4497643B2, OS:17.0.1, name:iPad Pro (11-inch) (4th generation) }
2025-01-06T14:22:09.3040840Z { platform:iOS Simulator, id:107CB696-AB9E-4BFC-AD00-B4E6F1470A0E, OS:17.2, name:iPad Pro (11-inch) (4th generation) }
2025-01-06T14:22:09.3043660Z { platform:iOS Simulator, id:6BF7D7A9-9E8A-43D1-80EF-BD6301CB6762, OS:17.0.1, name:iPad Pro (12.9-inch) (6th generation) }
2025-01-06T14:22:09.3048010Z { platform:iOS Simulator, id:91F85002-0495-4077-95EB-138A4B31BD9F, OS:17.2, name:iPad Pro (12.9-inch) (6th generation) }
2025-01-06T14:22:09.3050600Z { platform:iOS Simulator, id:9F5160B4-4E52-4E3A-BB97-8C67539E7EC5, OS:17.0.1, name:iPad mini (6th generation) }
2025-01-06T14:22:09.3052530Z { platform:iOS Simulator, id:2841CD68-372A-4C67-BE9F-730FA7D2E190, OS:17.2, name:iPad mini (6th generation) }
2025-01-06T14:22:09.3054290Z { platform:iOS Simulator, id:F86512C4-7BD2-4E0B-954A-6B5CD96E85A3, OS:17.0.1, name:iPhone 14 }
2025-01-06T14:22:09.3055870Z { platform:iOS Simulator, id:C68FA801-0220-4C45-B7A0-61F007527C24, OS:17.2, name:iPhone 14 }
2025-01-06T14:22:09.3057510Z { platform:iOS Simulator, id:9D7E5346-AE28-4536-BDF2-6D8AC9D96666, OS:17.0.1, name:iPhone 14 Plus }
2025-01-06T14:22:09.3059190Z { platform:iOS Simulator, id:CDFA0365-5A29-469E-8A51-D04402ED69BD, OS:17.2, name:iPhone 14 Plus }
2025-01-06T14:22:09.3060870Z { platform:iOS Simulator, id:0E716B60-116E-4C3B-8DB0-313A0726A827, OS:17.0.1, name:iPhone 14 Pro }
2025-01-06T14:22:09.3062520Z { platform:iOS Simulator, id:6C0517D9-8023-4D1C-9E4E-E643F233581E, OS:17.2, name:iPhone 14 Pro }
2025-01-06T14:22:09.3064470Z { platform:iOS Simulator, id:2D291A0B-763B-4C36-995F-9E351D48D6AC, OS:17.0.1, name:iPhone 14 Pro Max }
2025-01-06T14:22:09.3066460Z { platform:iOS Simulator, id:B13FFED6-0D1F-476D-99DC-F4DE09A93B77, OS:17.2, name:iPhone 14 Pro Max }
2025-01-06T14:22:09.3068360Z { platform:iOS Simulator, id:A33305FD-6CBC-4AD4-B5E3-65CEF85B6969, OS:17.0.1, name:iPhone 15 }
2025-01-06T14:22:09.3069990Z { platform:iOS Simulator, id:116D1E30-F4F0-4737-BFDB-746F38A77AF8, OS:17.2, name:iPhone 15 }
2025-01-06T14:22:09.3071650Z { platform:iOS Simulator, id:F8DBD11C-F466-4CA4-921B-01AC9BD4BF21, OS:17.0.1, name:iPhone 15 Plus }
2025-01-06T14:22:09.3073320Z { platform:iOS Simulator, id:849D7D2A-B6B5-4878-B10E-F612D0405D83, OS:17.2, name:iPhone 15 Plus }
2025-01-06T14:22:09.3075150Z { platform:iOS Simulator, id:044DC2BD-D34B-4B49-AF4F-F11574B6D782, OS:17.0.1, name:iPhone 15 Pro }
2025-01-06T14:22:09.3076960Z { platform:iOS Simulator, id:4DE5EECA-62A1-4E24-93E9-3295810361B5, OS:17.2, name:iPhone 15 Pro }
2025-01-06T14:22:09.3078850Z { platform:iOS Simulator, id:22486642-C0C6-41D0-A3AF-C80C2DC16425, OS:17.0.1, name:iPhone 15 Pro Max }
2025-01-06T14:22:09.3081150Z { platform:iOS Simulator, id:2A4BA5E4-684C-4336-9CA1-5BB2F27CA9A7, OS:17.2, name:iPhone 15 Pro Max }
2025-01-06T14:22:09.3083480Z { platform:iOS Simulator, id:FB316C94-655B-4F52-94EF-75AEC7ABAAD7, OS:17.0.1, name:iPhone SE (3rd generation) }
2025-01-06T14:22:09.3085520Z { platform:iOS Simulator, id:D83FEAB7-55E8-49C8-9E5D-697943752185, OS:17.2, name:iPhone SE (3rd generation) }
2025-01-06T14:22:09.3088330Z 2025-01-06 14:22:09.228 xcodebuild[90237:646906] Writing error result bundle to /var/folders/9f/9p4dh6hs5yddrk7drxq8rc_80000gn/T/ResultBundle_2025-06-01_14-22-0009.xcresult
2025-01-06T14:22:10.6893120Z xcodebuild: error: Failed to build workspace RNTesterPods with scheme RNTester.: Cannot test target “RNTesterIntegrationTests” on “Any iOS Device”: Tests must be run on a concrete device
2025-01-06T14:22:10.7820250Z Cannot test target “RNTesterUnitTests” on “Any iOS Device”: Tests must be run on a concrete device
2025-01-06T14:22:10.7821260Z 
2025-01-06T14:22:10.8824480Z ----- xcbeautify -----
2025-01-06T14:22:10.9128370Z Version: 2.16.0
2025-01-06T14:22:10.9143300Z ----------------------
2025-01-06T14:22:10.9343840Z 
2025-01-06T14:22:10.9478740Z ##[error]Signing for "RNTester" requires a development team. Select a development team in the Signing & Capabilities editor. (in target 'RNTester' from project 'RNTesterPods')
```

cipolleschi commented:
if you rebase on main once again, the problem should be fixed. Unit tests needs an emulator to run and I changed the CI to build for any emulator as we want also the RNTester App to be downloadable for other reasons...

kraenhansen committed:
Add "run-unit-tests" to the matrix only for hermes, new-arch and release

kraenhansen commented:
I've rebased and pushed with a commit adding `run-unit-tests` as a matrix variant, enabling this only when:
```
jsengine: Hermes
architecture: NewArch
flavor: Release
```

kraenhansen commented:
[test_ios_rntester (Hermes, NewArch, Release)](https://github.com/facebook/react-native/actions/runs/12817671055/job/35741686378?pr=48499#logs) is failing with an undefined symbols error while linking:

```
2025-01-16T21:49:19.6677370Z ld: Undefined symbols:
2025-01-16T21:49:19.6679310Z   facebook::hermes::inspector_modern::RuntimeAdapter::~RuntimeAdapter(), referenced from:
2025-01-16T21:49:19.6681090Z       facebook::react::HermesInstanceRuntimeAdapter::~HermesInstanceRuntimeAdapter() in libReact-RuntimeHermes.a[x86_64][2](HermesInstance.o)
2025-01-16T21:49:19.6683440Z       facebook::react::(anonymous namespace)::HermesExecutorRuntimeAdapter::~HermesExecutorRuntimeAdapter() in libReact-hermes.a[x86_64][3](HermesExecutorFactory.o)
2025-01-16T21:49:19.6686350Z   facebook::hermes::inspector_modern::chrome::CDPHandler::registerCallbacks(std::__1::function<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&)>, std::__1::function<void ()>), referenced from:
2025-01-16T21:49:19.6691220Z       facebook::hermes::inspector_modern::chrome::ConnectionDemux::addPage(std::__1::shared_ptr<facebook::hermes::inspector_modern::chrome::CDPHandler>)::$_0::operator()(std::__1::unique_ptr<facebook::react::jsinspector_modern::IRemoteConnection, std::__1::default_delete<facebook::react::jsinspector_modern::IRemoteConnection>>) const in libReact-hermes.a[x86_64][2](ConnectionDemux.o)
2025-01-16T21:49:19.6694460Z   facebook::hermes::inspector_modern::chrome::CDPHandler::unregisterCallbacks(), referenced from:
2025-01-16T21:49:19.6696100Z       facebook::hermes::inspector_modern::chrome::ConnectionDemux::removePage(int) in libReact-hermes.a[x86_64][2](ConnectionDemux.o)
2025-01-16T21:49:19.6698140Z       facebook::hermes::inspector_modern::chrome::(anonymous namespace)::LocalConnection::disconnect() in libReact-hermes.a[x86_64][2](ConnectionDemux.o)
2025-01-16T21:49:19.6704010Z   facebook::hermes::inspector_modern::chrome::CDPHandler::create(std::__1::unique_ptr<facebook::hermes::inspector_modern::RuntimeAdapter, std::__1::default_delete<facebook::hermes::inspector_modern::RuntimeAdapter>>, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, bool, bool, std::__1::shared_ptr<facebook::hermes::inspector_modern::chrome::State>, facebook::hermes::inspector_modern::chrome::CDPHandlerSessionConfig const&, std::__1::optional<facebook::hermes::inspector_modern::chrome::CDPHandlerExecutionContextDescription>), referenced from:
2025-01-16T21:49:19.6711030Z       facebook::hermes::inspector_modern::chrome::ConnectionDemux::enableDebugging(std::__1::unique_ptr<facebook::hermes::inspector_modern::RuntimeAdapter, std::__1::default_delete<facebook::hermes::inspector_modern::RuntimeAdapter>>, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&) in libReact-hermes.a[x86_64][2](ConnectionDemux.o)
2025-01-16T21:49:19.6714630Z   facebook::hermes::inspector_modern::chrome::CDPHandler::handle(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>), referenced from:
2025-01-16T21:49:19.6717680Z       facebook::hermes::inspector_modern::chrome::(anonymous namespace)::LocalConnection::sendMessage(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>) in libReact-hermes.a[x86_64][2](ConnectionDemux.o)
2025-01-16T21:49:19.6720980Z   facebook::hermes::cdp::CDPDebugAPI::addConsoleMessage(facebook::hermes::cdp::ConsoleMessage), referenced from:
2025-01-16T21:49:19.6725100Z       facebook::react::jsinspector_modern::HermesRuntimeTargetDelegate::Impl::addConsoleMessage(facebook::jsi::Runtime&, facebook::react::jsinspector_modern::ConsoleMessage) in libReact-hermes.a[x86_64][5](HermesRuntimeTargetDelegate.o)
2025-01-16T21:49:19.6728630Z   facebook::hermes::cdp::CDPDebugAPI::create(facebook::hermes::HermesRuntime&, unsigned long), referenced from:
2025-01-16T21:49:19.6731530Z       facebook::react::jsinspector_modern::HermesRuntimeTargetDelegate::Impl::Impl(facebook::react::jsinspector_modern::HermesRuntimeTargetDelegate&, std::__1::shared_ptr<facebook::hermes::HermesRuntime>) in libReact-hermes.a[x86_64][5](HermesRuntimeTargetDelegate.o)
2025-01-16T21:49:19.6733880Z   facebook::hermes::cdp::CDPDebugAPI::~CDPDebugAPI(), referenced from:
2025-01-16T21:49:19.6735780Z       std::__1::default_delete<facebook::hermes::cdp::CDPDebugAPI>::operator()[abi:v160006](facebook::hermes::cdp::CDPDebugAPI*) const in libReact-hermes.a[x86_64][5](HermesRuntimeTargetDelegate.o)
2025-01-16T21:49:19.6737910Z   facebook::hermes::cdp::State::State(facebook::hermes::cdp::State&&), referenced from:
2025-01-16T21:49:19.6741530Z       std::__1::__unique_if<facebook::react::jsinspector_modern::HermesRuntimeAgentDelegate::Impl::HermesStateWrapper>::__unique_single std::__1::make_unique[abi:v160006]<facebook::react::jsinspector_modern::HermesRuntimeAgentDelegate::Impl::HermesStateWrapper, facebook::hermes::cdp::State>(facebook::hermes::cdp::State&&) in libReact-hermes.a[x86_64][4](HermesRuntimeAgentDelegate.o)
2025-01-16T21:49:19.6746250Z       facebook::react::jsinspector_modern::HermesRuntimeAgentDelegate::Impl::HermesStateWrapper::HermesStateWrapper(facebook::hermes::cdp::State) in libReact-hermes.a[x86_64][4](HermesRuntimeAgentDelegate.o)
2025-01-16T21:49:19.6750200Z       facebook::react::jsinspector_modern::HermesRuntimeAgentDelegate::Impl::HermesStateWrapper::unwrapDestructively(facebook::react::jsinspector_modern::RuntimeAgentDelegate::ExportedState*) in libReact-hermes.a[x86_64][4](HermesRuntimeAgentDelegate.o)
2025-01-16T21:49:19.6752460Z   facebook::hermes::cdp::State::State(), referenced from:
2025-01-16T21:49:19.6754970Z       facebook::react::jsinspector_modern::HermesRuntimeAgentDelegate::Impl::HermesStateWrapper::unwrapDestructively(facebook::react::jsinspector_modern::RuntimeAgentDelegate::ExportedState*) in libReact-hermes.a[x86_64][4](HermesRuntimeAgentDelegate.o)
2025-01-16T21:49:19.6758970Z       facebook::react::jsinspector_modern::HermesRuntimeAgentDelegate::Impl::HermesStateWrapper::unwrapDestructively(facebook::react::jsinspector_modern::RuntimeAgentDelegate::ExportedState*) in libReact-hermes.a[x86_64][4](HermesRuntimeAgentDelegate.o)
2025-01-16T21:49:19.6761210Z   facebook::hermes::cdp::State::~State(), referenced from:
2025-01-16T21:49:19.6763040Z       facebook::react::jsinspector_modern::HermesRuntimeAgentDelegate::Impl::getExportedState() in libReact-hermes.a[x86_64][4](HermesRuntimeAgentDelegate.o)
2025-01-16T21:49:19.6765460Z       facebook::react::jsinspector_modern::HermesRuntimeAgentDelegate::Impl::getExportedState() in libReact-hermes.a[x86_64][4](HermesRuntimeAgentDelegate.o)
2025-01-16T21:49:19.6769670Z       std::__1::__unique_if<facebook::react::jsinspector_modern::HermesRuntimeAgentDelegate::Impl::HermesStateWrapper>::__unique_single std::__1::make_unique[abi:v160006]<facebook::react::jsinspector_modern::HermesRuntimeAgentDelegate::Impl::HermesStateWrapper, facebook::hermes::cdp::State>(facebook::hermes::cdp::State&&) in libReact-hermes.a[x86_64][4](HermesRuntimeAgentDelegate.o)
2025-01-16T21:49:19.6775250Z       std::__1::__unique_if<facebook::react::jsinspector_modern::HermesRuntimeAgentDelegate::Impl::HermesStateWrapper>::__unique_single std::__1::make_unique[abi:v160006]<facebook::react::jsinspector_modern::HermesRuntimeAgentDelegate::Impl::HermesStateWrapper, facebook::hermes::cdp::State>(facebook::hermes::cdp::State&&) in libReact-hermes.a[x86_64][4](HermesRuntimeAgentDelegate.o)
2025-01-16T21:49:19.6779260Z       facebook::react::jsinspector_modern::HermesRuntimeAgentDelegate::Impl::HermesStateWrapper::~HermesStateWrapper() in libReact-hermes.a[x86_64][4](HermesRuntimeAgentDelegate.o)
2025-01-16T21:49:19.6786450Z       facebook::react::jsinspector_modern::HermesRuntimeAgentDelegate::Impl::Impl(std::__1::function<void (std::__1::basic_string_view<char, std::__1::char_traits<char>>)>, facebook::react::jsinspector_modern::SessionState&, std::__1::unique_ptr<facebook::react::jsinspector_modern::RuntimeAgentDelegate::ExportedState, std::__1::default_delete<facebook::react::jsinspector_modern::RuntimeAgentDelegate::ExportedState>>, facebook::react::jsinspector_modern::ExecutionContextDescription const&, facebook::hermes::HermesRuntime&, facebook::react::jsinspector_modern::HermesRuntimeTargetDelegate&, std::__1::function<void (std::__1::function<void (facebook::jsi::Runtime&)>&&)> const&) in libReact-hermes.a[x86_64][4](HermesRuntimeAgentDelegate.o)
2025-01-16T21:49:19.6797250Z       facebook::react::jsinspector_modern::HermesRuntimeAgentDelegate::Impl::Impl(std::__1::function<void (std::__1::basic_string_view<char, std::__1::char_traits<char>>)>, facebook::react::jsinspector_modern::SessionState&, std::__1::unique_ptr<facebook::react::jsinspector_modern::RuntimeAgentDelegate::ExportedState, std::__1::default_delete<facebook::react::jsinspector_modern::RuntimeAgentDelegate::ExportedState>>, facebook::react::jsinspector_modern::ExecutionContextDescription const&, facebook::hermes::HermesRuntime&, facebook::react::jsinspector_modern::HermesRuntimeTargetDelegate&, std::__1::function<void (std::__1::function<void (facebook::jsi::Runtime&)>&&)> const&) in libReact-hermes.a[x86_64][4](HermesRuntimeAgentDelegate.o)
2025-01-16T21:49:19.6802470Z       ...
2025-01-16T21:49:19.7025360Z   facebook::hermes::cdp::CDPAgent::handleCommand(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>), referenced from:
2025-01-16T21:49:19.7034980Z       facebook::react::jsinspector_modern::HermesRuntimeAgentDelegate::Impl::handleRequest(facebook::react::jsinspector_modern::cdp::PreparsedRequest const&) in libReact-hermes.a[x86_64][4](HermesRuntimeAgentDelegate.o)
2025-01-16T21:49:19.7038020Z   facebook::hermes::cdp::CDPAgent::enableRuntimeDomain(), referenced from:
2025-01-16T21:49:19.7051810Z       facebook::react::jsinspector_modern::HermesRuntimeAgentDelegate::Impl::Impl(std::__1::function<void (std::__1::basic_string_view<char, std::__1::char_traits<char>>)>, facebook::react::jsinspector_modern::SessionState&, std::__1::unique_ptr<facebook::react::jsinspector_modern::RuntimeAgentDelegate::ExportedState, std::__1::default_delete<facebook::react::jsinspector_modern::RuntimeAgentDelegate::ExportedState>>, facebook::react::jsinspector_modern::ExecutionContextDescription const&, facebook::hermes::HermesRuntime&, facebook::react::jsinspector_modern::HermesRuntimeTargetDelegate&, std::__1::function<void (std::__1::function<void (facebook::jsi::Runtime&)>&&)> const&) in libReact-hermes.a[x86_64][4](HermesRuntimeAgentDelegate.o)
2025-01-16T21:49:19.7059160Z   facebook::hermes::cdp::CDPAgent::enableDebuggerDomain(), referenced from:
2025-01-16T21:49:19.7068330Z       facebook::react::jsinspector_modern::HermesRuntimeAgentDelegate::Impl::Impl(std::__1::function<void (std::__1::basic_string_view<char, std::__1::char_traits<char>>)>, facebook::react::jsinspector_modern::SessionState&, std::__1::unique_ptr<facebook::react::jsinspector_modern::RuntimeAgentDelegate::ExportedState, std::__1::default_delete<facebook::react::jsinspector_modern::RuntimeAgentDelegate::ExportedState>>, facebook::react::jsinspector_modern::ExecutionContextDescription const&, facebook::hermes::HermesRuntime&, facebook::react::jsinspector_modern::HermesRuntimeTargetDelegate&, std::__1::function<void (std::__1::function<void (facebook::jsi::Runtime&)>&&)> const&) in libReact-hermes.a[x86_64][4](HermesRuntimeAgentDelegate.o)
2025-01-16T21:49:19.7078980Z   facebook::hermes::cdp::CDPAgent::create(int, facebook::hermes::cdp::CDPDebugAPI&, std::__1::function<void (std::__1::function<void (facebook::hermes::HermesRuntime&)>)>, std::__1::function<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&)>, facebook::hermes::cdp::State), referenced from:
2025-01-16T21:49:19.7109670Z       facebook::react::jsinspector_modern::HermesRuntimeAgentDelegate::Impl::Impl(std::__1::function<void (std::__1::basic_string_view<char, std::__1::char_traits<char>>)>, facebook::react::jsinspector_modern::SessionState&, std::__1::unique_ptr<facebook::react::jsinspector_modern::RuntimeAgentDelegate::ExportedState, std::__1::default_delete<facebook::react::jsinspector_modern::RuntimeAgentDelegate::ExportedState>>, facebook::react::jsinspector_modern::ExecutionContextDescription const&, facebook::hermes::HermesRuntime&, facebook::react::jsinspector_modern::HermesRuntimeTargetDelegate&, std::__1::function<void (std::__1::function<void (facebook::jsi::Runtime&)>&&)> const&) in libReact-hermes.a[x86_64][4](HermesRuntimeAgentDelegate.o)
2025-01-16T21:49:19.7116960Z   facebook::hermes::cdp::CDPAgent::getState(), referenced from:
2025-01-16T21:49:19.7118900Z       facebook::react::jsinspector_modern::HermesRuntimeAgentDelegate::Impl::getExportedState() in libReact-hermes.a[x86_64][4](HermesRuntimeAgentDelegate.o)
2025-01-16T21:49:19.7120480Z   facebook::hermes::cdp::CDPAgent::~CDPAgent(), referenced from:
2025-01-16T21:49:19.7122250Z       std::__1::default_delete<facebook::hermes::cdp::CDPAgent>::operator()[abi:v160006](facebook::hermes::cdp::CDPAgent*) const in libReact-hermes.a[x86_64][4](HermesRuntimeAgentDelegate.o)
2025-01-16T21:49:19.7124490Z   facebook::hermes::inspector_modern::chrome::CDPHandler::getTitle() const, referenced from:
2025-01-16T21:49:19.7128110Z       facebook::hermes::inspector_modern::chrome::ConnectionDemux::enableDebugging(std::__1::unique_ptr<facebook::hermes::inspector_modern::RuntimeAdapter, std::__1::default_delete<facebook::hermes::inspector_modern::RuntimeAdapter>>, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&) in libReact-hermes.a[x86_64][2](ConnectionDemux.o)
2025-01-16T21:49:19.7132390Z       facebook::hermes::inspector_modern::chrome::ConnectionDemux::removePage(int) in libReact-hermes.a[x86_64][2](ConnectionDemux.o)
2025-01-16T21:49:19.7136220Z       facebook::hermes::inspector_modern::chrome::ConnectionDemux::addPage(std::__1::shared_ptr<facebook::hermes::inspector_modern::chrome::CDPHandler>) in libReact-hermes.a[x86_64][2](ConnectionDemux.o)
2025-01-16T21:49:19.7143940Z       facebook::hermes::inspector_modern::chrome::(anonymous namespace)::LocalConnection::LocalConnection(std::__1::shared_ptr<facebook::hermes::inspector_modern::chrome::CDPHandler>, std::__1::shared_ptr<std::__1::unordered_set<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::hash<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>, std::__1::equal_to<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>>>) in libReact-hermes.a[x86_64][2](ConnectionDemux.o)
2025-01-16T21:49:19.7149160Z       facebook::hermes::inspector_modern::chrome::(anonymous namespace)::LocalConnection::disconnect() in libReact-hermes.a[x86_64][2](ConnectionDemux.o)
2025-01-16T21:49:19.7150940Z   facebook::hermes::debugger::Debugger::captureStackTrace() const, referenced from:
2025-01-16T21:49:19.7153190Z       facebook::react::jsinspector_modern::HermesRuntimeTargetDelegate::Impl::captureStackTrace(facebook::jsi::Runtime&, unsigned long) in libReact-hermes.a[x86_64][5](HermesRuntimeTargetDelegate.o)
2025-01-16T21:49:19.7155400Z   typeinfo for facebook::hermes::inspector_modern::RuntimeAdapter, referenced from:
2025-01-16T21:49:19.7158050Z       typeinfo for facebook::react::HermesInstanceRuntimeAdapter in libReact-RuntimeHermes.a[x86_64][2](HermesInstance.o)
2025-01-16T21:49:19.7160690Z       typeinfo for facebook::react::(anonymous namespace)::HermesExecutorRuntimeAdapter in libReact-hermes.a[x86_64][3](HermesExecutorFactory.o)
2025-01-16T21:49:19.7163260Z   vtable for facebook::hermes::inspector_modern::RuntimeAdapter, referenced from:
2025-01-16T21:49:19.7164980Z       facebook::hermes::inspector_modern::RuntimeAdapter::RuntimeAdapter() in libReact-RuntimeHermes.a[x86_64][2](HermesInstance.o)
2025-01-16T21:49:19.7167660Z clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

cipolleschi commented:
Main is green... do you mind rebasing on top of it?

kraenhansen commented:
I've rebased on latest `main`.

kraenhansen commented:
The tests are failing now due to this change, introduced four days ago with https://github.com/facebook/react-native/pull/48755. I've added a comment on the PR with [a suggested change](https://github.com/facebook/react-native/pull/48755/files#r1923930396).

I can also include that fix in this PR if you'd want me to?

cipolleschi commented:
We are making several changes in this infrastructure, so I expect for some back and forth here. :(

kraenhansen committed:
Fix NewArchitectureHelper.new_arch_enabled

kraenhansen committed:
Update fixtures

cortinico commented:
Closing as not relevant anymore

cortinico commented:
> @cortinico interesting, because the PR is outdated or the code is going away? I still don't see the unit tests being enabled in the test-all workflow on `main` 🤔

Ah you're right. I've overlooked this PR as I thought it was related to E2E tests (which we change significantly over the last months), while instead it's related to iOS unit tests, so we can reopen it.
